From b82122c7e36d6a819ce8d469a93ca7a4001be257 Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Tue, 11 Jan 2022 15:09:36 +0530
Subject: [PATCH] sigma dut : make h2e as default sae password element method

WFA R2 test case 4.77.1 checks whether HASH_TO_ELEMENT method is used
as password element(PWE) in SAE.

WPA3 mandates H2E to be used as password element method for sae.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 ap.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ap.c b/ap.c
index cca7417..3fe8639 100644
--- a/ap.c
+++ b/ap.c
@@ -2451,6 +2451,7 @@ static enum sigma_cmd_result cmd_ap_set_security(struct sigma_dut *dut,
 			dut->ap_key_mgmt = AP_WPA2_SAE;
 			dut->ap_cipher = AP_CCMP;
 			dut->ap_pmf = AP_PMF_REQUIRED;
+			dut->sae_pwe = SAE_PWE_H2E;
 		} else if (strcasecmp(val, "WPA2-PSK-SAE") == 0) {
 			dut->ap_key_mgmt = AP_WPA2_PSK_SAE;
 			dut->ap_cipher = AP_CCMP;
-- 
2.7.4

